server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java

```diff
@@ -10,63 +16,56 @@
 import io.micronaut.http.client.HttpClient;
 import io.micronaut.http.client.annotation.Client;
 import io.micronaut.http.client.exceptions.HttpClientResponseException;
-import io.micronaut.test.annotation.MicronautTest;
-import io.micronaut.test.annotation.MockBean;
+
 import org.junit.jupiter.api.Test;
 
 import javax.inject.Inject;
 import java.util.*;
-import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
+import static com.objectcomputing.checkins.services.role.RoleType.Constants.MEMBER_ROLE;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.*;
 
-@MicronautTest
-class AgendaItemControllerTest {
+
+class AgendaItemControllerTest extends TestContainersSuite implements MemberProfileFixture, CheckInFixture, AgendaItemFixture {
 
     @Inject
     @Client("/services/agenda-item")
     HttpClient client;
-    @Inject
-    private AgendaItemServices agendaItemServices;
 
-    @MockBean(AgendaItemServices.class)
-    public AgendaItemServices agendaItemServices() {
-        return mock(AgendaItemServices.class);
-    }
 
     @Test
     void testCreateAnAgendaItem() {
-        AgendaItemCreateDTO agendaItemCreateDTO = new AgendaItemCreateDTO();
-        agendaItemCreateDTO.setCheckinid(UUID.randomUUID());
-        agendaItemCreateDTO.setCreatedbyid(UUID.randomUUID());
-        agendaItemCreateDTO.setDescription("dnc");
 
-        AgendaItem a = new AgendaItem(agendaItemCreateDTO.getCheckinid(), agendaItemCreateDTO.getCreatedbyid(), agendaItemCreateDTO.getDescription());
+        MemberProfile memberProfile = createADefaultMemberProfile();
+        MemberProfile memberProfileForPDL = createADefaultMemberProfileForPdl(memberProfile);
+
+        CheckIn checkIn = createADefaultCheckIn(memberProfile,memberProfileForPDL);
```

Comment:
`createADeafultAgendaItem()` expects a checkin and memberprofile so this need to be changed.